packages/-ember-data/tests/integration/relationships/belongs-to-test.js

```diff
@@ -119,6 +119,99 @@ module('integration/relationship/belongs-to BelongsTo Relationships (new-style)'
     assert.ok(personPet === pet, 'We ended up in the same state');
   });
 
+  test('async belongsTo resolves to the record when sideloaded', async function(assert) {
+    let personFindRecordCalls = 0;
+    this.owner.register(
+      'adapter:pet',
+      JSONAPIAdapter.extend({
+        shouldBackgroundReloadRecord() {
+          return false;
+        },
+        findRecord() {
+          assert.ok(false, 'We should not attempt to fetch the pet');
+          // still resolve so that test can continue nicely
+          return resolve({
+            data: {
+              type: 'pet',
+              id: '1',
+              attributes: { name: 'Shen' },
+              relationships: {
+                bestHuman: {
+                  data: { type: 'person', id: '1' },
+                },
+              },
+            },
+          });
+        },
+        findBelongsTo() {
+          assert.ok(false, 'We should not attempt to fetch the person for out pet');
+          return this.store.adapterFor('person').findRecord();
+        },
+      })
+    );
+
+    this.owner.register(
+      'adapter:person',
+      JSONAPIAdapter.extend({
+        shouldBackgroundReloadRecord() {
+          return false;
+        },
+        findRecord() {
+          assert.equal(++personFindRecordCalls, 1, 'We call findRecord only once for our person');
+          return resolve({
+            data: {
+              type: 'person',
+              id: '1',
+              attributes: { name: 'Chris' },
+              relationships: {
+                bestDog: {
+                  data: { type: 'pet', id: '1' },
+                  links: {
+                    related: './pet/1',
+                  },
+                },
+              },
+            },
+            included: [
+              {
+                type: 'pet',
+                id: '1',
+                attributes: { name: 'Shen' },
+                relationships: {
+                  bestHuman: {
+                    data: { type: 'person', id: '1' },
+                  },
+                },
+              },
+            ],
+          });
+        },
+        findBelongsTo() {
+          return this.store.adapterFor('pet').findRecord();
+        },
+      })
+    );
+
+    const person = await store.findRecord('person', '1');
+    const pet = store.peekRecord('pet', '1');
+    const personPetRequest = person.get('bestDog');
+    const personPet = await personPetRequest;
+
+    assert.ok(personPet === pet, 'We ended up in the correct state via .get');
+
+    let promiseDog;
+    await personPetRequest.then(dog => {
```

Comment:
Can you write a comment about what the goal of this test is, I'm pretty confused when reading it. Doesn't await return the resolved value anyhow, so we have the `personPet` from before?

Reply:
The test description is clear... `async belongsTo resolves to the record when sideloaded`

This test checks that various mechanisms of accessing an async belongsTo correctly resolve to the record instance (vs a promiseProxy) when the record was sideloaded. It additionally guards against accidental network requests to ensure that we are actually accessing the sideloaded version.

Reply:
I added some additional comments and specified in the title that we are resolving to the record, not a proxy to make it even more clear :)

Reply:
Thanks that helped a lot